### DISCUSSION TITLE
GPU performance on MOM6

### DISCUSSION THREAD

**Discussion by Kihang-Youn** [NOAA-GFDL/MOM6#721] [General]
Hi team,
I thought this would be a good place to start, so I'm reaching out to ask. If there's a better place to inquire, please let me know. 
Could I get an update on the progress of accelerating mom6 or mom5 with OpenACC (or OpenMP or CUDA)? 
I only have the previous results from NCAR, but is there any update?
https://www.cisl.ucar.edu/sites/default/files/2022-09/Dickerson_James_Stack_Slides.pdf

**Comment by marshallward** [comment]
Porting MOM6 to GPUs is a very high priority at the moment, but it is also a very new effort.  We have started to make some progress, but it will be some time until it can be used in production.  We've begun some positive discussions with other partners, so we're hoping that things will soon accelerate. :sunglasses:

Regarding the barotropic GPU acceleration project that you referenced: We're aware of the work but have not been updated on its progress, and no PRs have been sent to the repository.  You might have more luck reaching out to NCAR, or perhaps even the authors.  But at this time, the work is not yet part of MOM6.

  * NCAR MOM6 Forums: https://bb.cgd.ucar.edu/cesm/forums/mom6.148/
  * NCAR MOM6 repository: https://github.com/NCAR/MOM6/

As mentioned, this effort is in early stages, so any and all support is greatly appreciated.

This is a rather large topic that could go in many directions, so I am going to migrate it to "Discussions".

**Comment by marshallward** [comment]
This discussion has been moved to https://github.com/mom-ocean/MOM6/discussions/1638
